src/login/pam_systemd.c

```diff
@@ -374,37 +405,87 @@ _public_ PAM_EXTERN int pam_sm_open_session(
                 return PAM_SESSION_ERR;
         }
 
-        if (debug)
+        if (debug) {
                 pam_syslog(handle, LOG_DEBUG, "Asking logind to create session: "
                            "uid="UID_FMT" pid="PID_FMT" service=%s type=%s class=%s desktop=%s seat=%s vtnr=%"PRIu32" tty=%s display=%s remote=%s remote_user=%s remote_host=%s",
                            pw->pw_uid, getpid_cached(),
                            strempty(service),
                            type, class, strempty(desktop),
                            strempty(seat), vtnr, strempty(tty), strempty(display),
                            yes_no(remote), strempty(remote_user), strempty(remote_host));
+                pam_syslog(handle, LOG_DEBUG, "Session limits: "
+                           "memory_max=%s tasks_max=%s cpu_weight=%s io_weight=%s",
+                           strempty(memory_max), strempty(tasks_max), strempty(cpu_weight), strempty(io_weight));
+        }
+
+        r = sd_bus_message_new_method_call(
+                        bus,
+                        &m,
+                        "org.freedesktop.login1",
+                        "/org/freedesktop/login1",
+                        "org.freedesktop.login1.Manager",
+                        "CreateSession");
+        if (r < 0)
+                return r;
+
+        r = sd_bus_message_append(m, "uusssssussbss",
+                        (uint32_t) pw->pw_uid,
+                        (uint32_t) getpid_cached(),
+                        service,
+                        type,
+                        class,
+                        desktop,
+                        seat,
+                        vtnr,
+                        tty,
+                        display,
+                        remote,
+                        remote_user,
+                        remote_host);
+        if (r < 0)
+                return r;
+
+        r = sd_bus_message_open_container(m, 'a', "(sv)");
+        if (r < 0)
+                return r;
+
+        if (!isempty(memory_max)) {
+                r = parse_size(memory_max, 1024, &val);
```

Comment:
we should really support "infinity" here too. Sooner or later people will be tripped up by this...

In fact, I think this should be much closer to the behaviour of MemoryMax= parsing on the "systemctl set-property" command line. i.e. see:

https://github.com/systemd/systemd/blob/master/src/shared/bus-unit-util.c#L434

This stuff also permits setting percentage values, and I figure we want this here too.

Ideally we'd generalize that parsing code actually, and use it in both set-property handling and here, but that's not as easy as its sounds, since PAM modules want very different error logging, through pam_syslog(), hence I figure it might make sense to replicate a minimal version here that uses the right syntax.